src/main/java/io/antmedia/datastore/db/DataStore.java

```diff
@@ -22,208 +37,1017 @@
 import io.antmedia.datastore.db.types.Token;
 import io.antmedia.datastore.db.types.VoD;
 import io.antmedia.datastore.db.types.WebRTCViewerInfo;
+import io.antmedia.muxer.IAntMediaStreamHandler;
+import io.antmedia.muxer.MuxAdaptor;
 
 public abstract class DataStore {
 
-
-	//Do not forget to write function descriptions especially if you are adding new functions
+	// Do not forget to write function descriptions especially if you are adding new
+	// functions
 
 	public static final int MAX_ITEM_IN_ONE_LIST = 250;
+	private static final String REPLACE_CHARS_REGEX = "[\n|\r|\t]";
 
 	private boolean writeStatsToDatastore = true;
 
 	protected volatile boolean available = false;
 
 	protected static Logger logger = LoggerFactory.getLogger(DataStore.class);
-	
-	
-	public abstract String save(Broadcast broadcast);
+
+	public String save(Broadcast broadcast) {
+		return save(null, null, null, null, broadcast, null);
+	}
+
+	public String save(RMap<String, String> redisBroadcastMap, BTreeMap<String, String> mapdbBroadcastMap,
```

Comment:
Please don't have references to subclass in master class. In other words, there should be nothing about MapDB(BTreeMap) , Redis(RMap) or any other thing related to the subclass in this class